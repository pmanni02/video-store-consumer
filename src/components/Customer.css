.customer {
  margin: 2em;
  padding: 5px;
  width: 10em;
  background-color: pink;
}

.customer img {
  padding-top: 2em;
}

.customer img:hover {
  cursor: pointer;
}
