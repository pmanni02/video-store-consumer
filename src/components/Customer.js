import React, { Component } from 'react';
import PropTypes from 'prop-types';
import profile from './profile.jpg'

class Customer extends Component {

  rentalDetail = () => {
    this.props.customerCallback(this.props.id, this.props.name)
  }

  render() {
    return (
      <article>
        <div>{this.props.id}</div>
        <img src={profile} onClick={this.rentalDetail}/>
        <p>{this.props.name}</p>
        <p>{this.props.address}</p>
        <p>{this.props.city}, {this.props.state} {this.props.postal_code}</p>
        <p>{this.props.phone}</p>
      </article>
    )
  }
}

export default Customer;

Customer.propTypes = {
  id: PropTypes.number,
  name: PropTypes.string,
  address: PropTypes.string,
  city: PropTypes.string,
  state: PropTypes.string,
  postal_code: PropTypes.string,
  phone: PropTypes.string,
  customerCallback: PropTypes.func,
}
