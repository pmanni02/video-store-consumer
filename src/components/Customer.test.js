import React from 'react';
import { mount } from 'enzyme';
import Customer from './Customer';

describe('Board', () => {
  test('snapshot', () => {
    const wrapper = mount(
      <Customer
        id={0}
        name=''
        address=''
        city=''
        state=''
        postal_code=''
        phone=''
        customerCallback={()=>{}}
      />
    );
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });
});
