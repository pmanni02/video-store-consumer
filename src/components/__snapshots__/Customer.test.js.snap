// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board snapshot 1`] = `
<Customer
  address=""
  city=""
  customerCallback={[Function]}
  id={0}
  name=""
  phone=""
  postal_code=""
  state=""
>
  <article
    className="customer"
  >
    <img
      alt="customer profile"
      onClick={[Function]}
      src="profile.jpg"
    />
    <h2 />
    <p />
    <p>
      , 
       
    </p>
    <p />
  </article>
</Customer>
`;
